packages/hierarchies-react/src/test/internal/TreeModel.test.ts

```diff
@@ -0,0 +1,795 @@
+
+      TreeModel.setHierarchyLimit(model, "root-1", "unbounded");
+      expect(getHierarchyNode(model, "root-1")?.hierarchyLimit).to.be.eq("unbounded");
+    });
+
+    it("removes limit", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1"],
+        },
+        {
+          id: "root-1",
+          hierarchyLimit: 100,
+        },
+      ]);
+
+      expect(getHierarchyNode(model, "root-1")?.hierarchyLimit).to.be.eq(100);
+
+      TreeModel.setHierarchyLimit(model, "root-1", undefined);
+      expect(getHierarchyNode(model, "root-1")?.hierarchyLimit).to.be.undefined;
+    });
+
+    it("sets `isLoading` = `true` when new limit is set on expanded node", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1"],
+        },
+        {
+          id: "root-1",
+          isExpanded: true,
+          isLoading: false,
+        },
+      ]);
+
+      expect(getHierarchyNode(model, "root-1")?.hierarchyLimit).to.be.undefined;
+      expect(getHierarchyNode(model, "root-1")?.isLoading).to.be.false;
+
+      expect(TreeModel.setHierarchyLimit(model, "root-1", 100)).to.be.true;
+      expect(getHierarchyNode(model, "root-1")?.hierarchyLimit).to.be.eq(100);
+      expect(getHierarchyNode(model, "root-1")?.isLoading).to.be.true;
+    });
+
+    it("removes subtree when limit is set", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1"],
+        },
+        {
+          id: "root-1",
+          children: ["child-1"],
+        },
+        {
+          id: "child-1",
+          children: ["child-1-1"],
+        },
+        {
+          id: "child-1-1",
+        },
+      ]);
+
+      expect(getHierarchyNode(model, "root-1")).to.not.be.undefined;
+      expect(getHierarchyNode(model, "child-1")).to.not.be.undefined;
+      expect(getHierarchyNode(model, "child-1-1")).to.not.be.undefined;
+
+      TreeModel.setHierarchyLimit(model, "root-1", 100);
+      expect(getHierarchyNode(model, "root-1")).to.not.be.undefined;
+      expect(getHierarchyNode(model, "child-1")).to.be.undefined;
+      expect(getHierarchyNode(model, "child-1-1")).to.be.undefined;
+    });
+
+    it("does nothing if node does not exist", () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1"],
+        },
+        {
+          id: "root-1",
+          hierarchyLimit: undefined,
+        },
+      ]);
+
+      TreeModel.setHierarchyLimit(model, "invalid", 100);
+      expect(getHierarchyNode(model, "root-1")?.hierarchyLimit).to.be.undefined;
+    });
+  });
+
+  describe("setInstanceFilter", () => {
+    it("sets filter on tree root", () => {
+      const model = createTreeModel([]);
+      const filter: GenericInstanceFilter = { rules: { operator: "and", rules: [] }, propertyClassNames: [], relatedInstances: [] };
```

Comment:
maybe define this under `setInstanceFilter` `describe` block and reuse in all tests?